server/channels/api4/post.go

```diff
@@ -341,7 +345,12 @@ func getPostsForChannelAroundLastUnread(c *Context, w http.ResponseWriter, r *ht
 	}
 
 	channelId := c.Params.ChannelId
-	if !c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), channelId, model.PermissionReadChannelContent) {
+	channel, err := c.App.GetChannel(c.AppContext, channelId)
+	if err != nil {
+		c.Err = err
```

Comment:
Here and in a few other places I'd recommend to use `appErr` as the variable name for consistency and also to avoid future interface conversion bugs.

Reply:
It seems like most of this file we're using `err` for `model.AppError` returning functions. I definitely agree we should use `appErr` for consistency, but to stay consistent with the file for now I'm going to leave this one alone.

This would be a good candidate for a community campaign or something :)